File: Binary Search/0462.Minimum Moves to Equal Array Elements II/Solution.java
class Solution {
    int[] nums;

    public int minMoves2(int[] nums) {
        this.nums = nums;
        int left = Integer.MAX_VALUE, right= Integer.MIN_VALUE;
        for (int v : nums) {
            left = Math.min(left, v);
            right = Math.max(right, v);
        }
        while (left < right) {
            int mid = (left + right) >> 1;
            int cnt1 = getDiff(mid);
            int cnt2 = getDiff(mid + 1);
            if (cnt2 < cnt1) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }
        return getDiff(left);
    }

    public int getDiff(int x) {
        int cnt = 0;
        for (int v : nums) {
            cnt += Math.abs(v - x);
        }
        return cnt;
    }
}

File: Binary Search/0462.Minimum Moves to Equal Array Elements II/Solution.py
class Solution:
    def minMoves2(self, nums: List[int]) -> int:
        def get_diff(x):
            nonlocal nums
            cnt = 0
            for v in nums:
                cnt += abs(v - x)
            return cnt

        left, right = min(nums), max(nums)
        while left < right:
            mid = (left + right) >> 1
            cnt1 = get_diff(mid)
            cnt2 = get_diff(mid + 1)
            if cnt2 < cnt1:
                left = mid + 1
            else:
                right = mid
        return get_diff(left)